fix(transformer/class-properties): `private_field_count` is incorrect when class only have `accessor` with private field

Dunqing opened the pull request:
```ts
abstract class Foo {
  accessor prop: number = 1;
  static accessor prop2: number = 1;
  accessor #prop3: number = 1;
  accessor [prop4]: number = 1;
  private accessor prop5: number = 1;
  abstract accessor prop6: number;
}
```
This case only has a `#prop3` private field with an accessor. Since we haven't supported transforming `accessor` yet, the `class_details.is_transform_required` is always `false` in this `class`, which caused the `private_field_count` to be incorrect. Moving the decremental part to the `pop` side so that we can make sure the `private_props_prop` can always be handled correctly.

Dunqing commented:
* **#10463** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/10463?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/10463?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`
<hr />

## How to use the Graphite Merge Queue

Add either label to this PR to merge it via the merge queue:
- <kbd>0-merge</kbd> - adds this PR to the back of the merge queue
- <kbd>hotfix</kbd> - for urgent hot fixes, skip the queue and merge this PR next

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.dev/invite/github/oxc-project?ref=merge-queue-instructions-in-stack-comment&prId=10463).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

overlookmotel commented:
### Merge activity

* **Apr 17, 6:14 AM EDT**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

Dunqing committed:
fix(transformer/class-properties): `private_field_count` is incorrect when class only have `accessor` with private field (#10463)

```ts
abstract class Foo {
  accessor prop: number = 1;
  static accessor prop2: number = 1;
  accessor #prop3: number = 1;
  accessor [prop4]: number = 1;
  private accessor prop5: number = 1;
  abstract accessor prop6: number;
}
```
This case only has a `#prop3` private field with an accessor. Since we haven't supported transforming `accessor` yet, the `class_details.is_transform_required` is always `false` in this `class`, which caused the `private_field_count` to be incorrect. Moving the decremental part to the `pop` side so that we can make sure the `private_props_prop` can always be handled correctly.
